gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsRecord.java

```diff
@@ -0,0 +1,290 @@
+ * limitations under the License.
+ */
+
+package com.google.gcloud.dns;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import java.io.Serializable;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A class that represents a Google Cloud DNS record set.
+ *
+ * <p>A {@code DnsRecord} is the unit of data that will be returned by the DNS servers upon a DNS
+ * request for a specific domain. The {@code DnsRecord} holds the current state of the DNS records
+ * that make up a managed zone. You can read the records but you do not modify them directly.
+ * Rather, you edit the records in a managed zone by creating a ChangeRequest.
+ *
+ * @see <a href="https://cloud.google.com/dns/api/v1/resourceRecordSets">Google Cloud DNS
+ * documentation</a>
+ */
+public class DnsRecord implements Serializable {
+
+  private static final long serialVersionUID = 2016011914302204L;
+  private final String name;
+  private final List<String> rrdatas;
+  private final Integer ttl;
+  private final DnsRecordType type;
+
+  /**
+   * Enum for the DNS record types supported by Cloud DNS.
+   *
+   * <p>Google Cloud DNS currently supports records of type A, AAAA, CNAME, MX NAPTR, NS, PTR, SOA,
+   * SPF, SRV, TXT.
+   *
+   * @see <a href="https://cloud.google.com/dns/what-is-cloud-dns#supported_record_types">Cloud DNS
+   * supported record types</a>
+   */
+  public enum DnsRecordType {
+    /**
+     * Address record, which is used to map host names to their IPv4 address.
+     */
+    A,
+    /**
+     * IPv6 Address record, which is used to map host names to their IPv6 address.
+     */
+    AAAA,
+    /**
+     * Canonical name record, which is used to alias names.
+     */
+    CNAME,
+    /**
+     * Mail exchange record, which is used in routing requests to mail servers.
+     */
+    MX,
+    /**
+     * Naming authority pointer record, defined by RFC3403.
+     */
+    NAPTR,
+    /**
+     * Name server record, which delegates a DNS zone to an authoritative server.
+     */
+    NS,
+    /**
+     * Pointer record, which is often used for reverse DNS lookups.
+     */
+    PTR,
+    /**
+     * Start of authority record, which specifies authoritative information about a DNS zone.
+     */
+    SOA,
+    /**
+     * Sender policy framework record, which is used in email validation systems.
+     */
+    SPF,
+    /**
+     * Service locator record, which is used by some voice over IP, instant messaging protocols and
+     * other applications.
+     */
+    SRV,
+    /**
+     * Text record, which can contain arbitrary text and can also be used to define machine readable
+     * data such as security or abuse prevention information.
+     */
+    TXT
+  }
+
+  public static class Builder {
```

Comment:
I know we are not consistent with it and that it is fairly trivial but a javadoc simply stating that this class is a Builder
of {@code DnsRecord} would still be nice.